intake_dal/tests/test_dal_online.py

```diff
@@ -27,33 +26,52 @@ def user_events_df():
     )
 
 
+@pytest.fixture
+def user_events_json():
+    return [{
+        "userid": 100,
+        "home_id": 3,
+        "action": "click",
+        "timestamp": {
+            "format": "DATETIME",
+            "time": "2012-05-01T00:00:00.000000000",
+        }
+    },
+        {
+            "userid": 101,
+            "home_id": 4,
+            "action": "click",
+            "timestamp": {
+                "format": "DATETIME",
+                "time": "2012-05-02T00:00:00.000000000",
+            }
+        }
+    ]
+
+
 @mock.patch("intake_dal.dal_online._http_put_avro_data_set")
 @mock.patch("intake_dal.dal_online._http_get_avro_data_set")
 def test_dal_online_write_read(
-    mock_get: MagicMock, mock_put: MagicMock, serving_cat: DalCatalog, user_events_df: pd.DataFrame
+        mock_get: MagicMock, mock_put: MagicMock, serving_cat: DalCatalog, user_events_df: pd.DataFrame,
+        user_events_json: List[Dict]
 ):
     canonical_name = "entity.user.user_events"
     avro_str = serialize_panda_df_to_str(
         user_events_df, schema=json.loads(serving_cat.metadata["data_schema"][canonical_name])
     )
-    mock_get.return_value = avro_str
+    mock_get.return_value = user_events_json
     mock_put.return_value = 200
 
     serving_cat.entity.user.user_events(storage_mode="serving").write(user_events_df)
     assert serving_cat.entity.user.user_events(key=100).read().iloc[0].userid == 100
     assert serving_cat.entity.user.user_events(key=100).read().iloc[0].home_id == 3
 
-    assert_frame_equal(user_events_df, serving_cat.entity.user.user_events(key=100).read(), check_dtype=False)
```

Comment:
I could have removed line 42 and 44 here and use only `assert_frame_equal()`. I think that's cleaner.

Reply:
agree! Fixed